base/file.jl

```diff
@@ -458,6 +458,32 @@ function tempdir()
     end
 end
 
+"""
+    prepare_for_deletion(path::AbstractString)
+
+Prepares the given `path` for deletion by ensuring that all directories within that
+`path` have write permissions, so that files can be removed from them.  This is
```

Comment:
write _and traversal_ permissions

Do we not need read permissions?

Reply:
I think we've just never encountered a directory without read permissions in practice.  Sounds reasonable to me though.